channeldb/db.go

```diff
@@ -240,6 +242,28 @@ func Open(dbPath string, modifiers ...OptionModifier) (*DB, error) {
 	return db, err
 }
 
+func resetChanStateCache(cache *kvdb.Cache) error {
+	cache.Wipe()
+
+	if err := cache.AddTopLevelBucket(nodeBucket); err != nil {
+		return err
+	}
+
+	if err := cache.AddTopLevelBucket(edgeBucket); err != nil {
+		return err
+	}
+
+	if err := cache.AddTopLevelBucket(edgeIndexBucket); err != nil {
```

Comment:
IIRC, we already have a cache here there that was added to speed up the gossip queries...

Reply:
Yes I kept it there for the moment. If we end up liking the ideas in this PR I'll remove that. For now we need all top-level buckets to be cached since we do not read on-demand if the cache is empty for a bucket.